ProjFS.Mac/PrjFSKextTests/HandleOperationTests.mm

```diff
@@ -0,0 +1,553 @@
+            nullptr));
+    XCTAssertTrue(
+       MockCalls::DidCallFunction(
+            ProviderMessaging_TrySendRequestAndWaitForResponse,
+            _,
+            MessageType_KtoU_NotifyFilePreConvertToFull,
+            testFileVnode.get(),
+            _,
+            _,
+            _,
+            _,
+            _,
+            nullptr));
+}
+
+-(void) testWriteFileFull {
+    testFileVnode->attrValues.va_flags = FileFlags_IsInVirtualizationRoot;
+    HandleVnodeOperation(
+        nullptr,
+        nullptr,
+        KAUTH_VNODE_WRITE_DATA,
+        reinterpret_cast<uintptr_t>(context),
+        reinterpret_cast<uintptr_t>(testFileVnode.get()),
+        0,
+        0);
+    XCTAssertFalse(
+       MockCalls::DidCallFunction(
+            ProviderMessaging_TrySendRequestAndWaitForResponse,
+            _,
+            MessageType_KtoU_HydrateFile,
+            testFileVnode.get(),
+            _,
+            _,
+            _,
+            _,
+            _,
+            nullptr));
+    XCTAssertFalse(
+       MockCalls::DidCallFunction(
+            ProviderMessaging_TrySendRequestAndWaitForResponse,
+            _,
+            MessageType_KtoU_NotifyFilePreConvertToFull,
+            testFileVnode.get(),
+            _,
+            _,
+            _,
+            _,
+            _,
+            nullptr));
+}
+
+- (void) testEventsAreIgnored {
+    testDirVnode->attrValues.va_flags = FileFlags_IsInVirtualizationRoot | FileFlags_IsEmpty;
+    const int actionCount = 5;
+    kauth_action_t actions[actionCount] =
+    {
+        KAUTH_VNODE_APPEND_DATA,
+        KAUTH_VNODE_ADD_SUBDIRECTORY,
+        KAUTH_VNODE_WRITE_SECURITY,
+        KAUTH_VNODE_TAKE_OWNERSHIP,
+        KAUTH_VNODE_ACCESS
+    };
+    
+    for (int i = 0; i < actionCount; i++)
+    {
+        // Verify for File node
+        HandleVnodeOperation(
+            nullptr,
+            nullptr,
+            actions[i],
+            reinterpret_cast<uintptr_t>(context),
+            reinterpret_cast<uintptr_t>(testFileVnode.get()),
+            0,
+            0);
+        XCTAssertFalse(
+            MockCalls::DidCallFunction(ProviderMessaging_TrySendRequestAndWaitForResponse));
+        MockCalls::Clear();
+
+        // Verify for Directory node
+        HandleVnodeOperation(
+            nullptr,
+            nullptr,
+            actions[i],
+            reinterpret_cast<uintptr_t>(context),
+            reinterpret_cast<uintptr_t>(testDirVnode.get()),
+            0,
+            0);
+        XCTAssertFalse(MockCalls::DidCallFunction(ProviderMessaging_TrySendRequestAndWaitForResponse));
+        MockCalls::Clear();
+    }
+}
+
+- (void) testOpen {
+    // A file that is not yet in the virtual root, should recieve a creation request
+    char pathBuffer[PrjFSMaxPath] = "";
+    int pathLength = sizeof(pathBuffer);
+    vn_getpath(testFileVnode.get(), pathBuffer, &pathLength);
+
+    HandleFileOpOperation(
+        nullptr,
+        nullptr,
+        1,
```

Comment:
Instead of 1 I want t use KAUTH_FILEOP_OPEN
I see this in Kernel/sys/kauth.h

@pmj whats the best way to import it into the unit tests here.

Reply:
It seems that `#include <Kernel/sys/kauth.h>` works with a bit of trickery. I've pushed a commit [to this branch](https://github.com/pmj/VFSForGit/tree/more_testing_handlevnodeop) that does that and avoids the need for our own `#define`s.

Reply:
@pmj Nice Thank You!